velox/exec/AggregateFunctionAdapterUtil.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/exec/Aggregate.h"
+#include "velox/expression/FunctionSignature.h"
+
+namespace facebook::velox::exec {
+
+std::unordered_map<std::string, SignatureVariable> getUsedTypeVariables(
+    const std::vector<TypeSignature>& types,
+    const std::unordered_map<std::string, SignatureVariable>& allVariables);
+
+class CompanionSignatures {
+ public:
+  static std::vector<AggregateFunctionSignaturePtr> partialFunctionSignatures(
+      const std::vector<AggregateFunctionSignaturePtr>& aggregateSignatures) {
+    std::vector<AggregateFunctionSignaturePtr> signatures;
+    for (const auto& signature : aggregateSignatures) {
+      std::vector<TypeSignature> usedTypes = signature->argumentTypes();
+      usedTypes.push_back(signature->intermediateType());
+      auto variables = getUsedTypeVariables(usedTypes, signature->variables());
+
+      signatures.push_back(std::make_shared<AggregateFunctionSignature>(
+          variables,
+          signature->intermediateType(),
+          signature->intermediateType(),
+          signature->argumentTypes(),
+          signature->constantArguments(),
+          signature->variableArity()));
+    }
+    return signatures;
+  }
+
+  static std::string partialFunctionName(const std::string& name) {
+    return name + "_partial";
+  }
+
+  static std::vector<AggregateFunctionSignaturePtr> mergeFunctionSignatures(
+      const std::vector<AggregateFunctionSignaturePtr>& aggregateSignatures) {
+    std::unordered_set<TypeSignature> distinctIntermediateTypes;
+    std::vector<AggregateFunctionSignaturePtr> signatures;
+    for (const auto& signature : aggregateSignatures) {
```

Comment:
Added below code for count_merge, because count allows empty input. But `std::vector<TypeSignature>{signature->intermediateType()}` has a size of 1.
```
      if (signature->constantArguments().size() == 0) {
        // For count_merge, the input cannot be empty.
        continue;
      }
```